fix(snippet): setting end_right_gravity

TheBlob42 opened the pull request:
The value for `end_right_gravity` is always set to true which causes issues with tabstops that are right next to each other

#### Reproduction

- execute `:lua vim.snippet.expand('${1:first}${2:second}')`
- change first stop to `FIRST`
- jump to second tabstop with `<tab>`
- change second stop to `SECOND`
- jump back to first tabstop with `<s-tab>`
- change stop to `first`

##### Expected

`firstSECOND`

##### Actual

`first`

![snippet_1_problem_high_res](https://github.com/user-attachments/assets/896df66a-746a-4a93-8372-62b271290f73)

#### Solution

Set the value for `end_right_gravity` to the opposite of `right_gravity`:

- deactivate tabstop expansion
    - `right_gravity = true` so writing before the tabstop will move it to the right
    - `end_right_gravity = false` so that writing on the end extmark will not expand it
- activate tabstop expansion
    - `right_gravity = false` so writing before the tabstop will be included
    - `end_right_gravity = true` so that writing on the end extmark will be included

![snippet_1_solution](https://github.com/user-attachments/assets/bcfcd552-4995-4f9a-bb07-e340b83323b2)

TheBlob42 commented:
@MariaSolOs I did directly edit the test you mentioned, as I think this "special case" fits the test case of adjacent tabstops perfectly. But if you think this should be a dedicated test I will change it accordingly

MariaSolOs commented:
> @MariaSolOs I did directly edit the test you mentioned, as I think this "special case" fits the test case of adjacent tabstops perfectly. But if you think this should be a dedicated test I will change it accordingly

That's fine with me!

Are you still working on the PR? If not, publish it so that we can merge it :shipit:

TheBlob42 commented:
From my end the PR is ready for review. I am just unsure why the adapted test is failing in some environments?

MariaSolOs commented:
> From my end the PR is ready for review. I am just unsure why the adapted test is failing in some environments?

Hmm that's interesting. Could you try adding a `poke_eventloop()` call after `feed('<S-Tab>')`?

TheBlob42 committed:
fix(snippet): setting end_right_gravity

When right_gravity is set to true for deactivating tabstop expansion we
have to set end_right_gravity to false to avoid expanding the tabstop
region on the right side. Vice versa for activating tabstop expansion
again.
